server/gitrest/packages/gitrest-base/src/utils/helpers.ts

```diff
@@ -119,18 +120,30 @@ export async function retrieveLatestFullSummaryFromStorage(
 /**
  * Retrieves the full repository path. Or throws an error if not valid.
  */
-export function getRepoPath(name: string, owner?: string): string {
-    // `name` needs to be always present and valid.
-    if (!name || path.parse(name).dir !== "") {
-        throw new NetworkError(400, `Invalid repo name provided.`);
+export function getRepoPath(
+    repoPerDocEnabled: boolean,
+    tenantId: string,
+    documentId: string,
+    owner?: string): string {
+    // `tenantId` needs to be always present and valid.
+    if (!tenantId || path.parse(tenantId).dir !== "") {
+        throw new NetworkError(400, `Invalid repo name (tenantId) provided.`);
     }
 
     // When `owner` is present, it needs to be valid.
     if (owner && path.parse(owner).dir !== "") {
         throw new NetworkError(400, `Invalid repo owner provided.`);
     }
 
-    return owner ? `${owner}/${name}` : name;
+    if (repoPerDocEnabled) {
+        // `documentId` needs to be always present and valid.
+        if (!documentId || path.parse(documentId).dir !== "") {
+            throw new NetworkError(400, `Invalid repo name (documentId) provided.`);
+        }
+        return owner ? `${owner}/${tenantId}/${documentId}` : `${tenantId}/${documentId}`;
+    }
+
+    return owner ? `${owner}/${tenantId}` : tenantId;
```

Comment:
(nit) I think this whole conditional return sequence could be simplified (if documentId is made optional and is undefined when repoPerDocEnabled==false)

```ts
return [owner, tenantId, documentId].filter(x => x !== undefined).join("/");
```

Since Array length===3, filter is performant enough to not have to worry about it

Reply:
Good idea :) Thanks for the suggestion.